addons/stock_account/models/stock_move.py

```diff
@@ -499,17 +497,12 @@ def _get_out_move_lines(self, lot=None):
         :returns: a subset of `self` containing the outgoing records
         :rtype: recordset
         """
-        res = self.env['stock.move.line']
-        for move_line in self.move_line_ids:
-            if lot and move_line.lot_id != lot:
-                continue
-            if not move_line.picked:
-                continue
-            if move_line._should_exclude_for_valuation():
-                continue
+        res_ids = []
+        move_lines = self.move_line_ids.filtered(lambda ml: ml.picked and (not lot or ml.lot_id == lot) and not ml._should_exclude_for_valuation())
```

Comment:
Why not a single `filtered` then?
Also, now you're browsing twice some records, as you browse them once in the first `filtered`, then once again in the following `for` loop.